doc/usecases/layers_alternatives.md

```diff
@@ -0,0 +1,30 @@
+# Use Case: Layer Alternatives
```

Comment:
I prefer the "versions" as a name for the feature because it would semantically mean that I could build and revert something incrementally like migrations. However, this is just a side note and is not a strong preference of mine compared to the actual implementation details

Reply:
In general, everything looks sharp and could be merged. Thank you once again!

Reply:
> I prefer the "versions" as a name for the feature because it would semantically mean that I could build and revert something incrementally like migrations. However, this is just a side note and is not a strong preference of mine compared to the actual implementation details

I think there is a misunderstanding of this feature. (Your term definition of versions gave me the indication about this misunderstanding.) I agree with you about the term: versions usually describe linear history. But in this usecase it is *not* about linear history.

If you don't like the term "alternatives", other fitting words would be scenarios, branches, forks, ...

But we need two different terms, as we have two different concepts here.